### PR TITLE
WT-7646 Remove unneeded uses of WT_WITH_BUCKET_STORAGE macro.

### PR DESCRIPTION
**Pull request by ddanderson**

### CODE REVIEW

**Review by sueloverso — APPROVED** [review]
LGTM as in the changes themselves are fine. I do not know when you do/do not need to use the macro however to talk to the correctness of that.

File: src/btree/bt_io.c [hunk 1]
```diff
@@ -35,16 +35,12 @@ __wt_bt_read(WT_SESSION_IMPL *session, WT_ITEM *buf, const uint8_t *addr, size_t
      * into the caller's buffer. Else, read directly into the caller's buffer.
      */
     if (btree->compressor == NULL && btree->kencryptor == NULL) {
-        WT_WITH_BUCKET_STORAGE(
-          btree->bstorage, session, { ret = bm->read(bm, session, buf, addr, addr_size); });
-        WT_RET(ret);
```
Comment (ddanderson) [inline src/btree/bt_io.c:40]: Some explanation...  This was saying: use the bucket storage/file system in `btree->bstorage` when you do file system operations.  But the file handles are cached and opened lazily, so this single operation doesn't necessarily help.  We'd have to encapsulate every read and write into the block manager in these calls to make this work.  Far better to just catch this at the point of the lazy open.  Which is what we did -- WT_FILE_OPENER handles that.  So these macros don't really do anything (any effect they have to switch the file system is redone at the point of the lazy open).  Therefore they can all be removed (except at the lazy open).

File: src/meta/meta_turtle.c [hunk 2]
```diff
@@ -129,9 +129,7 @@ __metadata_load_bulk(WT_SESSION_IMPL *session)
         WT_ERR(cursor->get_value(cursor, &value));
         filecfg[1] = value;
         WT_ERR(__wt_direct_io_size_check(session, filecfg, "allocation_size", &allocsize));
-        WT_WITH_BUCKET_STORAGE(
-          NULL, session, ret = __wt_block_manager_create(session, key, allocsize));
-        WT_ERR(ret);
```
Comment (ddanderson) [inline src/meta/meta_turtle.c:134]: OTOH, this is saying, use a NULL bucket storage, or just the default file system when creating this file.  It's a no-op, so can be removed.

**Comment by ddanderson** [comment]
@sueloverso and @keitharnoldsmith, Since it wasn't very clear, I've put some review annotation in the changes to (I hope) explain why the code can be removed.

**Comment by sueloverso** [comment]
Thanks for the explanation! That makes sense now. I (still) approve.

**Review by keitharnoldsmith — APPROVED** [review]
LGTM.  Thanks for the explanations; they definitely helped!

**Pull request merged**